src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,177 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T> implements Set<T> , Comparable {
+
+    static Object[] elementData = new Object[10];
+    static int modCount = 0;
+
+    static Number left;
+    static Number right;
+    boolean isEmpty = true;
+
+    public static Range<Integer> of(Integer a, Integer b) {
+        Range<Integer> range = new Range<>();
+        range.clear();
+        left = a;
+        right = b;
+        for(Integer j = left.intValue(); j<=right.intValue(); j++){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Float> of(Float a, Float b) {
+        Range<Float> range = new Range<>();
+        range.clear();
+        left = a;
+        right = b;
+        for(Float j = left.floatValue(); j<=right.floatValue(); j=j+0.1f){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> characterCharacterFunction) {
```

Comment:
You caught for **Character** (custom type). 
You have to create an implementation for any custom type, not only for Character. Look at this solution for Character, you just have to create a more general solution.